content/kips/kip-29.md

```diff
@@ -0,0 +1,57 @@
+---
+kip: 29
+title: Copy Trading Account MVP
+status: Draft
+author: Troy (@Tburm)
+created: 2022-09-08
+---
+
+## Summary
+
+A proposal to implement a feature to allow traders to open copy trading accounts on Kwenta.
+
+## Abstract
+
+This proposal outlines a new feature on Kwenta which allows traders to open an account which copies the trades of another account. Traders will specify a cross-margin account and a list of assets to copy to create a new copy trading account. Once the account is created, keepers will listen to trades on the target account and rebalance the copy trading account to mirror market exposure.
+
+## Motivation
+
+1. Traders have expressed interest in copying trades of other accounts on Kwenta. Since futures market trades are all executed on-chain, traders have access to historical trading data for every other trading account.
+2. Traders want to promote others to copy trade them in exchange for a share of their profits.
+
+## Specification
+
+Terminology:
+- Copy trading account: A cross-margin (CM) enabled smart contract account which mirrors trades from another CM account 
+- Target account: The account being copied by a copy trading account
+
+The `MarginBase.sol` contract provides users with an interface to manage margin and futures positions across many markets in a single transaction. Keepers will use these functions to copy positions from the target account and execute trades on the copy trading account. To enable these functions the `MarginBase.sol` contract needs some new functionality:
+
+- Stores the address of the target account (`address targetAccount`.)
```

Comment:
Marko mentioned before he'd prefer the ability to control if traders can copy him or not, do you think this is something we should consider in this version? And to avoid changes to CM contract around enabling / disabling copy trading we could deploy a separate contract which maintains a record of traders that want to opt out of copy trading, could be opt in by default.

Reply:
Maybe the solution here is to store those addresses on the copy trading factory. It stores an "opt out list" and won't create new accounts if the address is present in that list. Users would need to submit a transaction to get added to this list.

This would handle a default opt-in while giving traders an opt-out.